asset-transfer-private-data/chaincode-go/private_asset_transfer.go

```diff
@@ -0,0 +1,497 @@
+		}
+		if ownerAppraisedValueHash == nil {
+			return fmt.Errorf("hash of appraised value for %s does not exist in collection %s", assetID, collectionOwner)
+		}
+
+		// get hash of buyers agreed to value
+	buyerAppraisedValueHash, err := ctx.GetStub().GetPrivateDataHash(collectionBuyer, assetID)
+		if err != nil {
+			return fmt.Errorf("failed to hash of appraised value from buyer collection %s: %s", collectionBuyer ,err.Error())
+		}
+		if buyerAppraisedValueHash == nil {
+			return fmt.Errorf("hash of appraised value for %s does not exist in collection %s", assetID, collectionBuyer)
+		}
+
+	// verify that the two hashes match
+	if !bytes.Equal(ownerAppraisedValueHash, buyerAppraisedValueHash) {
+		return fmt.Errorf("hash for appraised value for owner %x does not value for seller %x", ownerAppraisedValueHash, buyerAppraisedValueHash)
+	}
+
+	return nil
+}
+
+// DeleteAsset can be used by the owner of the asset to delete the asset
+func (s *SmartContract) DeleteAsset(ctx contractapi.TransactionContextInterface) error {
+
+	transMap, err := ctx.GetStub().GetTransient()
+		if err != nil {
+			return fmt.Errorf("Error getting transient: " + err.Error())
+		}
+
+	// Asset properties are private, therefore they get passed in transient field
+	transientDeleteJSON, ok := transMap["asset_delete"]
+		if !ok {
+			return fmt.Errorf("asset to delete not found in the transient map")
+		}
+
+	type assetDelete struct {
+			ID string `json:"asset_id"`
+		}
+
+	var assetDeleteInput assetDelete
+	err = json.Unmarshal(transientDeleteJSON, &assetDeleteInput)
+		if err != nil {
+			return fmt.Errorf("failed to unmarshal JSON: %s", err.Error())
+		}
+
+	if len(assetDeleteInput.ID) == 0 {
+		return fmt.Errorf("ID field must be a non-empty string")
+	}
+
+	valAsbytes, err := ctx.GetStub().GetPrivateData(assetCollection, assetDeleteInput.ID) //get the asset from chaincode state
+		if err != nil {
+			return fmt.Errorf("failed to read asset: %s", err.Error())
+		}
+		if valAsbytes == nil {
+			return fmt.Errorf("asset private details does not exist: %s",  assetDeleteInput.ID)
+		}
+
+	var assetToDelete Asset
+	err = json.Unmarshal([]byte(valAsbytes), &assetToDelete)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal JSON: %s", err.Error())
+	}
+
+	// delete the asset from state
+	err = ctx.GetStub().DelPrivateData(assetCollection, assetDeleteInput.ID)
+	if err != nil {
+		return fmt.Errorf("Failed to delete state:" + err.Error())
+	}
+
+	// Finally, delete private details of asset
+
+	ownerCollection, err := getCollectionName(ctx) // get owners collection
+
+	err = ctx.GetStub().DelPrivateData(ownerCollection, assetDeleteInput.ID) // delete the asset
+	if err != nil {
+			return err
+	}
+
+	return nil
+
+}
+
+// DeleteProposal can be used by a user of a rejected transfer properties to
+// remove the propsal from the asset collection and the org specific collection
```

Comment:
DeleteTranferAgreement can be used by a owner/buyer? user to remove the TransferAgreement agreed upon earlier 

This usecase is not clear to me, since this deletes TransferAgreement + PrivateData from ownerCollection.
That leaves the Asset in assetCollection, without appraisedValue.

Either remove this chaincode, or if it is still useful beyond `DeleteAsset`, then let us explain the usecase in README. If this require a larger discussion, please tackle it in different PR

Reply:
The buyer that wishes to withdraw a proposal could use this function to get rid of the agreement. But it can be used to remove the proposal by the owner since it writes to the other organizations collection. So maybe this can be removed. Easy to delete, so I will leave this in for now.